File: striver/S3L1/rotate_array.java
package S3L1;

import java.util.Scanner;

public class rotate_array {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int size = sc.nextInt();
        int[] arr = new int[size];
        for(int i=0;i<size;i++){
            arr[i] = sc.nextInt();
        }
        for(int i=0;i<size;i++){
            System.out.print(arr[i]+" ");
        }
        System.out.println();
        rotate(arr);
        for(int i=0;i<size;i++){
            System.out.print(arr[i]+" ");
        }
        System.out.println();
    }
    public static void rotate(int[] arr){
        int temp = arr[0];
        for(int i=0;i<arr.length-1;i++){
            arr[i+1]=arr[i];
        }
        arr[arr.length-1]=temp;
    }
}

File: striver/S3L1/rotate_array.py
def rotate(arr):
    temp = arr[0]
    for i in range(1,size):
        arr[i-1]=arr[i]
    arr[size-1]=temp
    
size = int(input())
arr = [0]*size
for i in range(0,size):
    arr[i] = int(input())
for i in range(0,size):
    print(arr[i],end=" ")
print()
rotate(arr)
for i in range(0,size):
    print(arr[i],end=" ")
